src/mon/FSCommands.cc

```diff
@@ -1074,3 +1055,30 @@ int FileSystemCommandHandler::_check_pool(
   return 0;
 }
 
+int FileSystemCommandHandler::is_op_allowed(
+    MonOpRequestRef op, FSMap &fsmap, const cmdmap_t& cmdmap,
+    std::stringstream &ss)
+{
+    string fs_name;
+    cmd_getval(cmdmap, "fs_name", fs_name);
+
```

Comment:
Do the `FSMap` copy here:

`FSMap fscopy = fsmap`

and add a comment explaining why (so you can filter the file systems).